src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -150,11 +150,24 @@ PtrCouplingData BaseCouplingScheme::addCouplingData(const mesh::PtrData &data, m
 {
   int             id = data->getID();
   PtrCouplingData ptrCplData;
-  if (!utils::contained(id, _allData)) { // data is not used by this coupling scheme yet, create new CouplingData
+  if (!utils::contained(id, _allMeshData)) { // data is not used by this coupling scheme yet, create new CouplingData
     ptrCplData = std::make_shared<CouplingData>(data, std::move(mesh), requiresInitialization, _extrapolationOrder);
-    _allData.emplace(id, ptrCplData);
+    _allMeshData.emplace(id, ptrCplData);
   } else { // data is already used by another exchange of this coupling scheme, use existing CouplingData
-    ptrCplData = _allData[id];
+    ptrCplData = _allMeshData[id];
+  }
+  return ptrCplData;
+}
+
+PtrCouplingData BaseCouplingScheme::addGlobalCouplingData(const mesh::PtrData &data, bool requiresInitialization)
+{
+  int             id = data->getID();
+  PtrCouplingData ptrCplData;
+  if (!utils::contained(id, _allGlobalData)) { // data is not used by this coupling scheme yet, create new GlobalCouplingData
+    ptrCplData = std::make_shared<CouplingData>(data, nullptr, requiresInitialization, _extrapolationOrder);
+    _allGlobalData.emplace(id, ptrCplData);
+  } else { // data is already used by another exchange of this coupling scheme, use existing GlobalCouplingData
+    ptrCplData = _allGlobalData[id];
   }
   return ptrCplData;
 }
```

Comment:
Can't we avoid this duplication by an additional argument `isGlobal` in `addCouplingData`?
I have the feeling there are more cases like this one.

Reply:
Agree. I've removed the following functions which were straightforward to get rid of:
- addGlobalCouplingData
- addGlobalDataToSend, addGlobalDataToReceive
- sendGlobalData, receiveGlobalData

There are still more opportunities to avoid duplication but they break things in other places. I've documented these in the follow up issue: #1716